components/amara/amara.app.mjs

```diff
@@ -0,0 +1,543 @@
+    team:	{
+      type: "string",
+      label: "Team",
+      description: "Team slug for the video or null to remove it from its team.",
+      optional: true,
+    },
+    videoUrl: {
+      type: "string",
+      label: "Video URL",
+      description: "The url for the video. Any url that Amara accepts will work here. You can send the URL for a file (e.g. `http:///www.example.com/my-video.ogv`), or a link to one of the accepted providers (YouTube, Vimeo).",
+    },
+    primaryAudioLanguageCode: {
+      type: "string",
+      label: "Primary audio language code",
+      description: "Language code for the main language spoken in the video.",
+      optional: true,
+      async options() {
+        const { languages } = await this.getLanguages();
+
+        return Object.keys(languages)
+          .map((key) => ({
+            label: languages[key],
+            value: key,
+          }));
+      },
+    },
+    project: {
+      type: "string",
+      label: "Project",
+      description: "Project slug for the video or null to put it in the default project",
+      optional: true,
+      async options({
+        prevContext, team,
+      }) {
+        const { url } = prevContext;
+        if (url === null || !team) {
+          return [];
+        }
+        const {
+          meta,
+          objects: projects,
+        } = await this.getTeamProjects({
+          team,
+        });
+        const options = projects.map((project) => ({
+          label: project.name,
+          value: project.slug,
+        }));
+        return {
+          options,
+          context: {
+            url: meta.next,
+          },
+        };
+      },
+    },
+    videoId: {
+      type: "string",
+      label: "Video ID",
+      description: "Video ID as the Amara video identifier",
+      async options({
+        prevContext, team,
+      }) {
+        const { url } = prevContext;
+
+        if (url === null) {
+          return [];
+        }
+
+        const {
+          meta,
+          objects: videos,
+        } = await this.listVideos({
+          url,
+          params: {
+            limit: 20,
+            team: team ?? null,
+            // order_by: constants.ORDER_BY.CREATED_DESC,
+          },
+        });
+
+        const options = videos.map((video) => ({
+          label: video.title,
+          value: video.id,
+        }));
+
+        return {
+          options,
+          context: {
+            url: meta.next,
+          },
+        };
+      },
+    },
+    language: {
+      type: "string",
+      label: "Language",
+      description: "Language code for the language of the subtitles",
+      async options({
+        videoId, prevContext,
+      }) {
+        const { url } = prevContext;
+
+        if (url === null) {
+          return [];
+        }
+
+        const {
+          meta,
+          objects: languages,
+        } =
+          await this.getVideoSubtitleLanguages({
+            url,
+            videoId,
```

Comment:
I would say just to return `[]` in case of error because for all cases the language need to exist in order to call the request. (`delete-subtitles, fetch-raw-subtitles, fetch-subtitles-data, get-subtitle-language-details, perform-action, update-subtitle-language`)